model/endpoint.go

```diff
@@ -0,0 +1,11 @@
+package model
+
+import "net"
+
+// Endpoint holds the network context of a node in the service graph.
+type Endpoint struct {
+	ServiceName string `json:"serviceName,omitempty"`
+	IPv4        net.IP `json:"ipv4,omitempty"`
+	IPv6        net.IP `json:"ipv6,omitempty"`
+	Port        int    `json:"port,omitempty"`
```

Comment:
Could `uint` will work better for this case?

Reply:
`uint16` would even be more correct... will update